drivers/usb/udc/udc_mcux_ehci.c

```diff
@@ -92,6 +97,50 @@ static int udc_mcux_control(const struct device *dev, usb_device_control_type_t
 	return 0;
 }
 
+/* Helper function to keep track of the activity state for the udc */
+static void udc_mcux_change_state(const struct device *dev, bool enabled, bool vbus_present)
+{
+	struct udc_mcux_data *data = udc_get_private(dev);
+
+	if (data->enabled == enabled && data->vbus_present == vbus_present) {
+		return;
+	}
+
+	if (vbus_present != data->vbus_present && enabled) {
+		udc_submit_event(data->dev,
+				 vbus_present ? UDC_EVT_VBUS_READY : UDC_EVT_VBUS_REMOVED, 0);
+	}
+	if (enabled && vbus_present) {
+		data->enabled = true;
+		data->vbus_present = true;
+
+		/*
+		 * Block PM when usb is active.
```

Comment:
Block PM from doing what? What is "usb is active"?

Reply:
Sorry for not being more clear, this was maybe a bad assumption on my part, but the only context this driver had from a usb interface point of view was rather the driver controller was enabled or if a VBUS signal is detected so I made the wrong assumption that this could only mean the controller is active, I have updated the comment.